packages/core/src/auth/connection.ts

```diff
@@ -219,17 +280,40 @@ export class ProfileStore {
 
     public async addProfile(id: string, profile: SsoProfile): Promise<StoredProfile<SsoProfile>>
     public async addProfile(id: string, profile: IamProfile): Promise<StoredProfile<IamProfile>>
+    @withTelemetryContext({ name: 'addProfile', class: profileStoreClassName })
     public async addProfile(id: string, profile: Profile): Promise<StoredProfile> {
-        return this.putProfile(id, this.initMetadata(profile))
+        return telemetry.auth_modifyConnection.run(async () => {
+            telemetry.record({
+                action: 'addProfile',
+                id,
+                source: asStringifiedStack(telemetry.getFunctionStack()),
+            })
+
+            const newProfile = this.initMetadata(profile)
+            telemetry.record(getTelemetryForProfile(newProfile))
+
+            return await this.putProfile(id, newProfile)
+        })
     }
 
+    @withTelemetryContext({ name: 'updateProfile', class: profileStoreClassName })
     public async updateProfile(id: string, profile: Profile): Promise<StoredProfile> {
-        const oldProfile = this.getProfileOrThrow(id)
-        if (oldProfile.type !== profile.type) {
-            throw new Error(`Cannot change profile type from "${oldProfile.type}" to "${profile.type}"`)
-        }
+        return telemetry.auth_modifyConnection.run(async () => {
+            telemetry.record({
```

Comment:
This records to whatever the "active span" is. In this case it is `auth_modifyConnection` but it leaves room for potential problems down the line if we were to refactor for example.

The safest way is to do `span.record()` since that targets this specific `auth_modifyConnection`. You just need to add `span` as an argument on line 301